File: libs/document-processor-next/src/__tests__/models/remove-if-present.ts
import { z } from 'zod';
import { withTypeBlueId } from '@blue-labs/language';

import { handlerContractBaseSchema } from '../../model/shared/index.js';

export const removeIfPresentSchema = withTypeBlueId('RemoveIfPresent')(
  handlerContractBaseSchema.extend({
    propertyKey: z.string(),
  }),
);

export type RemoveIfPresent = z.infer<typeof removeIfPresentSchema>;

File: references/java-sdk/src/test/java/blue/language/processor/model/RemoveIfPresent.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;

@TypeBlueId("RemoveIfPresent")
public class RemoveIfPresent extends HandlerContract {

    private String propertyKey;

    public String getPropertyKey() {
        return propertyKey;
    }

    public RemoveIfPresent propertyKey(String propertyKey) {
        this.propertyKey = propertyKey;
        return this;
    }

    public void setPropertyKey(String propertyKey) {
        this.propertyKey = propertyKey;
    }
}
